src/app/navbar/navbar.component.ts

```diff
@@ -10,8 +12,23 @@ export class NavbarComponent {
 
   @Input() user: User;
   @Output() logout: EventEmitter<void> = new EventEmitter<void>();
-  @Output() updateName: EventEmitter<void> = new EventEmitter<void>();
 
-  constructor() { }
+  constructor(public dialog: MatDialog) { }
+
+  updateName() {
+    const dialogRef = this.dialog.open(UpdateNameModalComponent, {
+      width: '250px',
```

Comment:
Let's make this modal a bit wider and also handle the controls width when we change the width from here.